cmd/thor/utils.go

```diff
@@ -424,56 +424,78 @@ type p2pComm struct {
 }
 
 func newP2PComm(ctx *cli.Context, repo *chain.Repository, txPool *txpool.TxPool, instanceDir string) (*p2pComm, error) {
+	// known peers will be loaded/stored from/in this file
+	peersCachePath := filepath.Join(instanceDir, "peers.cache")
+
 	configDir, err := makeConfigDir(ctx)
 	if err != nil {
 		return nil, err
 	}
+
 	key, err := loadOrGeneratePrivateKey(filepath.Join(configDir, "p2p.key"))
 	if err != nil {
 		return nil, errors.Wrap(err, "load or generate P2P key")
 	}
+
 	nat, err := nat.Parse(ctx.String(natFlag.Name))
 	if err != nil {
 		cli.ShowAppHelp(ctx)
 		return nil, errors.Wrap(err, "parse -nat flag")
 	}
 
-	opts := &p2psrv.Options{
-		Name:            common.MakeName("thor", fullVersion()),
-		PrivateKey:      key,
-		MaxPeers:        ctx.Int(maxPeersFlag.Name),
-		ListenAddr:      fmt.Sprintf(":%v", ctx.Int(p2pPortFlag.Name)),
-		BootstrapNodes:  fallbackBootstrapNodes,
-		RemoteBootstrap: remoteBootstrapList,
-		NAT:             nat,
-	}
+	var knowNodes p2psrv.Nodes
+	var restrictedPeerListEnabled bool
+	remoteBootstrapURL := remoteBootstrapList
 
-	peersCachePath := filepath.Join(instanceDir, "peers.cache")
+	// restricted peer list mode - no other peers are loaded
```

Comment:
Reframed the existing behaviour to be clear.
- If there is a restricted list ONLY load that list into the KnownNodes
- Otherwise : 
  - Load default or flag-supplied boot nodes, and append those to any cache-stored nodes